tests/scripts/thread-cert/Cert_7_1_01_BorderRouterAsLeader.py

```diff
@@ -139,27 +137,17 @@ def test(self):
 
         # Step 4 - DUT sends a MLE Child ID Response to Router1
         msg = leader_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
-        # Only check the network_data in the child id response,
-        # pass a network_data check function to check the detail
-        check_child_id_response(msg, network_data=CheckType.CONTAIN, \
-            network_data_detail_check_func=functools.partial(check_network_data_tlv, min_prefix_count=2, prefix_check_func=check_prefix))
+        check_child_id_response_full_data(msg)
```

Comment:
Looking from 7.x.x of the test plan, for the steps that check network data in Data Response or Child Id Response or Child Update Request, it is required either the num of prefixes, or the exact matching prefixes with specified BR Rloc16 value. It would be better to define a general check approach. Below is one proposal for your reference

```
class NetworkDataCheckType:
   PREFIX_NUM = 1
   PREFIX_CONTENT = 2

def check_child_id_response(command_msg, ..., 
	network_data=CheckType.CONTAIN, 
	network_data_check = ()，# (check_type, check_content)
       )

check_child_id_response（msg, network_data=CheckType.CONTAIN, network_data_check = （NetworkDataCheckType.PREFIX_NUM, 2))
check_child_id_response（msg, network_data=CheckType.CONTAIN, network_data_check = （NetworkDataCheckType.PREFIX_CONTENT, [('2001:2:0:1::/64', 0xFFFE), ('2001:2:0:2::/64', BRRloc16)]))
```

Thoughts?

Reply:
Yeah, that would be great.